components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2404,13 +2452,13 @@ private Group addDomainToUserMembers(Group group, String userStoreDomain) throws
             offset = 1;
         }
 
-        // If the results is less than the offset return an empty user list.
+        // If the results are less than the offset or the limit (count) equals to zero, return an empty user list.
         if (offset > users.length) {
             return new String[0];
         }
 
         // If the limit is zero, all the users needs to be returned after verifying the offset.
-        if (limit <= 0) {
+        if (limit == 0) {
```

Comment:
Does the negative case is handled prior to this method?

Reply:
It is handled by the charon endpoint. But validating here also would be good.
fixed: df77256698c74c829147f73d00e2f63979199c16